pkg/system/api/types.go

```diff
@@ -0,0 +1,12 @@
+package api
```

Comment:
where is this file coming from?

Reply:
It's defined here inline in the [old OS file](https://github.com/alexander-ding/csi-proxy/blob/015225affc1e207eb931fec055653a407c8c3688/pkg/os/system/api.go#L17). Since this is user facing, it should really get its own definition. I just copied over the struct from API so that the internal impl and the PKG have their own versions.

Reply:
is it possible to move this to pkg/system/types.go?

Reply:
My understanding is that it's bad practice to return non-exported types in exported functions/methods. For example, if a client wants to implement the API interface in pkg/system/api.go (for whatever reason; maybe mocking), then they would want to be able to specify the type. This wouldn't be possible if the type isn't exported.

Reply:
sure, I was trying to minimize what's visible to the client but as you mention there are practical reasons to have it like this